### DISCUSSION TITLE
Adding code generator for a programming language or library or framework

### DISCUSSION THREAD

**Discussion by animator** [foss42/apidash#80] [General]
Currently, API Dash supports code generation for the [following language/library](https://github.com/foss42/apidash/tree/main#code-generators).
Contributors are welcome to add new code generators to this project which can be for new programming languages or libraries or frameworks.

You can:
- Browse the current list of required codegens available [here](https://github.com/foss42/apidash/labels/codegen) and send across a PR linking the issue.
- Or, Create a new issue with title **`Adding codegen for <language> using <library or framework>`** and send across a PR linking the issue.
- Or, modify an existing codegen to remove any bugs & improve its functionality or code quality.

### Some important points to note when you are working on a codegen

- Current codegens in the project are located in `lib/codegen/`
- To setup API Dash in dev check out our [Contribution Guidelines](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md).
- Along with every codegen, you also need to submit the corresponding test file located in `test/codegen/`.
- Make sure `flutter run` does not fail.
- The test script must contain the expected code for all the cases provided in `test/request_models.dart`. Take a look at any existing codegen test file to understand how test case is provided for the corresponding codegen.
- The expected code should be properly formatted as per that particular language standard and you should manually run and verify each expected code (golden test) to make sure they produce the correct result.
- Make sure `flutter test` does not fail.
- Add the local setup instructions in your PR on how someone can test the generated code locally.

**Comment by sibghatullah78** [comment]
Hi sir i want to implement new code generator for a programming language in api dash .That is ver common in flutter community and is used by most of the developers .i have done step 1 for this i have chosen Language: Dart,
Library: dio  ,By adding support for dio, API Dash would cater to a broader audience of Dart and Flutter developers, providing them with more options for generating API integration code.what is your suggestion and outline to implement this feature .Thanks

**Comment by sibghatullah78** [comment]
Sir, i have checked that in code folder for code generetor in API dash there is already file present of dio.dart I just want to insure that is this implemented or have some issue .

**Reply by ashitaprasad** [reply]
You can verify the results and improve the code generator.

**Comment by sibghatullah78** [comment]
okay mam

**Comment by shaleen31** [comment]
Hello I want to include code generators for the swift and c#.

**Reply by ashitaprasad** [reply]
Sure, you can go ahead and raise a PR.

**Comment by ashitaprasad** [comment]
We have closed the following [codegen issues](https://github.com/foss42/apidash/issues?q=label%3A%22not+planned%22+is%3Aclosed+label%3Acodegen) as they are no longer in our roadmap.

**Reply by badnikhil** [reply]
> We have closed the following [codegen issues](https://github.com/foss42/apidash/issues?q=label%3A%22not+planned%22+is%3Aclosed+label%3Acodegen) as they are no longer in our roadmap.

why these issues are  closed as completed even though the related features haven't been added yet. Is the code generation feature no longer a priority, or are there other reasons behind this decision?

**Reply by animator** [reply]
Because AI can be leveraged for code generation so dashbot can support the remaining codegens

**Reply by badnikhil** [reply]
it have multiple drawbacks, especially in the early stages.
Coming to the point — just wanted to confirm: in Idea-10 from the ideas list, the last point mentions 'more code coverage'. Does that refer to the codegen feature, or am I misunderstanding it? My proposal includes significant work on that feature, so I wanted to check if it still makes sense to include it in my project.

**Reply by animator** [reply]
@badnikhil Read - https://www.atlassian.com/continuous-delivery/software-testing/code-coverage

**Reply by badnikhil** [reply]
Thanks for confirming.

Given that a chunk of my proposal focused on codegen enhancements, and those are no longer in scope — should I remove that section entirely and rework the proposal accordingly?

**Reply by animator** [reply]
You can rework accordingly.
